In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
import datetime as dt
import matplotlib
import matplotlib.pyplot as plt
import os
import io
import geopandas as gpd
import seaborn as sns
# suppress warning
import warnings
warnings.filterwarnings('ignore')

%matplotlib inline

pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)
pd.set_option('display.width', None)
# display all columns

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# **Step 1: get the svi nyc data**

In [ ]:
file_path1 = '/content/drive/My Drive/X999/NewYork_cencus.csv'

In [ ]:
svi_census = pd.read_csv(file_path1)

In [ ]:
svi_census.shape

(5394, 158)

In [ ]:
fips_list = list(svi_census.FIPS.unique())
# fips_list

In [ ]:
nyc_county_codes = {'36005', '36047', '36061', '36081', '36085'}

In [ ]:
nyc_tracts = [fips for fips in fips_list if str(fips)[:5] in nyc_county_codes]

In [ ]:
len(nyc_tracts)

2324

In [ ]:
nyc_tracts[:5]

[np.int64(36005000100),
 np.int64(36005000200),
 np.int64(36005000400),
 np.int64(36005001600),
 np.int64(36005001901)]

In [ ]:
svi_nyc = svi_census[svi_census['FIPS'].isin(nyc_tracts)].copy()

In [ ]:
svi_nyc.shape

(2324, 158)

In [ ]:
svi_nyc.columns = svi_nyc.columns.str.lower()
svi_nyc.shape

(2324, 158)

In [ ]:
svi_nyc.columns

Index(['st', 'state', 'st_abbr', 'stcnty', 'county', 'fips', 'location',
       'area_sqmi', 'e_totpop', 'm_totpop',
       ...
       'ep_asian', 'mp_asian', 'ep_aian', 'mp_aian', 'ep_nhpi', 'mp_nhpi',
       'ep_twomore', 'mp_twomore', 'ep_otherrace', 'mp_otherrace'],
      dtype='object', length=158)

# **Step 2: clean svi_census as svi_nta**

In [ ]:
link2 = '/content/drive/My Drive/X999/svi_cleaned.csv'

In [ ]:
svi_nta = pd.read_csv(link2)
svi_nta.shape

(204, 153)

In [ ]:
svi_nta.head()

fips     location  area_sqmi  e_totpop  m_totpop   e_hu  m_hu   e_hh  \
0  10001  ZCTA5 10001   0.623822     27004      1827  16975   831  14375   
1  10002  ZCTA5 10002   0.822292     76518      2894  39094  1241  36028   
2  10003  ZCTA5 10003   0.571603     53877      2579  30766   956  24987   
3  10004  ZCTA5 10004   0.455576      4579       926   2706   484   2123   
4  10005  ZCTA5 10005   0.072868      8801      1132   6272   438   4881   

   m_hh  e_pov150  m_pov150  e_unemp  m_unemp  e_hburd  m_hburd  e_nohsdp  \
0   782      5248       797      761      266     3314      531      1930   
1  1326     27908      2853     2833      574    14688     1367     18301   
2   936      6397      1171     1613      315     5445      853      1574   
3   394       169       101       11       16      130       84        32   
4   550       647       363      257      158      532      266       168   

   m_nohsdp  e_uninsur  m_uninsur  e_age65  m_age65  e_age17  m_age17  \
0       534        831        289     3428      432     2694      643   
1      1376       4074        766    17681     1287    10028     1549   
2       422       1282        404     8128      792     3866      718   
3        28         13         19      190      131      840      341   
4       135        106         96      158      129      924      395   

   e_disabl  m_disabl  e_sngpnt  m_sngpnt  e_limeng  m_limeng  e_minrty  \
0      2310       499       501       215      1381       405     13460   
1      9896      1062      2211       499     18393      1640     56964   
2      3604       634       278       171      1217       828     19778   
3       104        82         8        16        39        73      2009   
4        99        80       140       147        22        78      2730   

   m_minrty  e_munit  m_munit  e_mobile  m_mobile  e_crowd  m_crowd  e_noveh  \
0      2305    15840      898        15        23      389      135    12285   
1      3226    35725     1677        16        28     2461      449    29828   
2      3548    27261     1299         0        31      798      282    20035   
3      1102     2592      476         0        13      161       99     1548   
4      1506     6083      451         0        19      411      200     4503   

   m_noveh  e_groupq  m_groupq  ep_pov150  mp_pov150  ep_unemp  mp_unemp  \
0      840      2213       218       20.3        2.7       4.3       1.5   
1     1403      2090        39       36.8        3.5       7.6       1.4   
2      905     10199       194       14.3        2.5       4.7       0.9   
3      318        35        13        3.7        2.1       0.4       0.5   
4      549        50        30        7.4        4.0       3.4       2.0   

   ep_hburd  mp_hburd  ep_nohsdp  mp_nohsdp  ep_uninsur  mp_uninsur  ep_age65  \
0      23.1       3.5        9.1        2.4         3.1         1.0      12.7   
1      40.8       3.5       30.0        2.0         5.4         1.0      23.1   
2      21.8       3.3        4.2        1.2         2.4         0.7      15.1   
3       6.1       3.8        0.9        0.8         0.3         0.4       4.1   
4      10.9       5.3        2.5        2.0         1.2         1.0       1.8   

   mp_age65  ep_age17  mp_age17  ep_disabl  mp_disabl  ep_sngpnt  mp_sngpnt  \
0       1.6      10.0       2.1        8.6        1.9        3.5        1.5   
1       1.7      13.1       1.8       13.0        1.4        6.1        1.4   
2       1.4       7.2       1.2        6.7        1.2        1.1        0.7   
3       2.9      18.3       5.1        2.3        1.9        0.4        0.7   
4       1.5      10.5       3.8        1.1        0.9        2.9        3.0   

   ep_limeng  mp_limeng  ep_minrty  mp_minrty  ep_munit  mp_munit  ep_mobile  \
0        5.3        1.5       49.8        7.8      93.3       2.7        0.1   
1       24.7        2.0       74.4        3.1      91.4       3.2        0.0   
2        2.3        1.6       36.7        6.3      88.

In [ ]:
columns_to_keep = svi_nta.columns.to_list()

In [ ]:
svi_nyc.drop(columns = [col for col in svi_nyc.columns if col not in columns_to_keep], inplace = True)
svi_nyc.shape

(2324, 153)

In [ ]:
svi_nyc.head()

fips                                    location  area_sqmi  \
101  36005000100      Census Tract 1; Bronx County; New York   0.647574   
102  36005000200      Census Tract 2; Bronx County; New York   0.173014   
103  36005000400      Census Tract 4; Bronx County; New York   0.299390   
104  36005001600     Census Tract 16; Bronx County; New York   0.187290   
105  36005001901  Census Tract 19.01; Bronx County; New York   0.079154   

     e_totpop  m_totpop  e_hu  m_hu  e_hh  m_hh  e_pov150  m_pov150  e_unemp  \
101      4446       616     0    13     0    13         0        13        0   
102      4870       621  1487   209  1425   212       910       382      108   
103      6257       982  2344   278  2309   281       648       246      285   
104      6177       618  2205   236  2205   236      1883       534      343   
105      2181       258   978   100   940   108       870       234       77   

     m_unemp  e_hburd  m_hburd  e_nohsdp  m_nohsdp  e_uninsur  m_uninsur  \
101       13        0       26      1495       291          0         13   
102       85      345      141       531       204        337        273   
103      174      855      293       892       468        175        117   
104      155     1186      330      1142       291        208        125   
105       61      532      140       246        89        259        203   

     e_age65  m_age65  e_age17  m_age17  e_disabl  m_disabl  e_sngpnt  \
101       31       24       58       57         0        13         0   
102      812      186      822      222       790       241        81   
103      828      431     1383      371       883       305       259   
104     1269      332     1201      249      1268       350       146   
105       19       21      475      140       429       120       164   

     m_sngpnt  e_limeng  m_limeng  e_minrty  m_minrty  e_munit  m_munit  \
101        18       363       120      3348       739        0       18   
102        69       614       311      4787       627        0       18   
103       119       508       449      5974      1002       28       57   
104        84       459       181      6071       620     1601      272   
105        56        24        59      1875       294      703      121   

     e_mobile  m_mobile  e_crowd  m_crowd  e_noveh  m_noveh  e_groupq  \
101         0        13        0       18        0       13      4446   
102         0        13       89       68      301      124         7   
103         0        19       72       73      600      312        19   
104         0        19      403      148     1193      276       225   
105         0        13       41       43      699      124        17   

     m_groupq  ep_pov150  mp_pov150  ep_unemp  mp_unemp  ep_hburd  mp_hburd  \
101       616     -999.0     -999.0    -999.0    -999.0    -999.0    -999.0   
102         1       18.7        7.5       4.5       3.6      24.2       9.2   
103         2       10.4        3.6       8.3       4.9      37.0      11.9   
104         9       31.6        8.3      11.9       4.7      53.8      13.8   
105         4       39.9        9.6       6.2       4.9      56.6      13.4   

     ep_nohsdp  mp_nohsdp  ep_uninsur  mp_uninsur  ep_age65  mp_age65  \
101       38.7        5.3      -999.0      -999.0       0.7       0.5   
102       14.8        5.2         6.9         5.3      16.7       4.5   
103       19.5        8.2         2.8         1.8      13.2       6.2   
104       27.6        6.6         3.5         2.1      20.5       4.8   
105       16.8        6.1        11.9         8.8       0.9       1.0   

     ep_age17  mp_age17  ep_disabl  mp_disabl  ep_sngpnt  mp_sngpnt  \
101       1.3       1.3     -999.0     -999.0     -999.0     -999.0   
102      16.9       4.1       16.2        5.1        5.7        4.8   
103      22.1       4.7       14.1        5.7       11.3        5.0   
104      19.4       3.4       21.2        5.4        6.6        3.7   
105      21.8       6.0  

In [ ]:
svi_census_pop = svi_nyc[['fips', 'location', 'area_sqmi', 'e_totpop']]
svi_census_pop

fips                                        location  area_sqmi  \
101   36005000100          Census Tract 1; Bronx County; New York   0.647574   
102   36005000200          Census Tract 2; Bronx County; New York   0.173014   
103   36005000400          Census Tract 4; Bronx County; New York   0.299390   
104   36005001600         Census Tract 16; Bronx County; New York   0.187290   
105   36005001901      Census Tract 19.01; Bronx County; New York   0.079154   
...           ...                                             ...        ...   
4282  36085030301  Census Tract 303.01; Richmond County; New York   0.277731   
4283  36085030302  Census Tract 303.02; Richmond County; New York   0.350875   
4284  36085031901  Census Tract 319.01; Richmond County; New York   0.149827   
4285  36085031902  Census Tract 319.02; Richmond County; New York   0.266618   
4286  36085032300     Census Tract 323; Richmond County; New York   1.493411   

      e_totpop  
101       4446  
102       4870  
103       6257  
104       6177  
105       2181  
...        ...  
4282      5915  
4283      6374  
4284      3674  
4285      5053  
4286      1133  

[2324 rows x 4 columns]

# **Step 3: Get the bbl_evictions_311_svi merged df**

In [ ]:
link3 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_normal_times.csv'
merged_df = pd.read_csv(link3)
merged_df.shape

(66397, 94)

In [ ]:
merged_df.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

In [ ]:
merged_df.census_tract

0         1152.0
1          118.0
2          122.0
3          409.0
4           35.0
          ...   
66392        7.0
66393      151.0
66394      151.0
66395       97.0
66396    11402.0
Name: census_tract, Length: 66397, dtype: float64

In [ ]:
merged_df['tract_str'] = merged_df['census_tract'].astype(str).str.replace('.0', '').str.zfill(6)
merged_df.tract_str

0        001152
1        000118
2        000122
3        000409
4        000035
          ...  
66392    000007
66393    000151
66394    000151
66395    000097
66396    011402
Name: tract_str, Length: 66397, dtype: object

In [ ]:
svi_census['tract_str'] = svi_census['FIPS'].astype(str).str[-6:]
svi_census.tract_str

0       000100
1       000201
2       000202
3       000301
4       000302
         ...  
5389    150301
5390    150302
5391    150400
5392    150501
5393    150502
Name: tract_str, Length: 5394, dtype: object

In [ ]:
overlapping_tracts = set(merged_df['tract_str']).intersection(set(svi_census['tract_str']))
num_overlapping = len(overlapping_tracts)
num_overlapping

355

In [ ]:
svi_census_pop.shape[0]

2324

In [ ]:
len(merged_df['tract_str'].unique())

1258

In [ ]:
nyc_county_codes = ['36005', '36047', '36061', '36081', '36085']
svi_nyc = svi_census[svi_census['FIPS'].astype(str).str[:5].isin(nyc_county_codes)].copy()

In [ ]:
merged_df['tract_str'] = merged_df['census_tract'].astype(str).str.replace('.0', '').str.zfill(6)

In [ ]:
svi_nyc['tract_str'] = svi_nyc['FIPS'].astype(str).str[-6:]

In [ ]:
overlap = set(merged_df['tract_str']).intersection(set(svi_nyc['tract_str']))
len(overlap) / len(merged_df)

0.005150835128093137

In [ ]:
missing = set(merged_df['tract_str']) - set(svi_nyc['tract_str'])
len(missing)

916

In [ ]:
merged_df.fips

0        11207
1        11220
2        11220
3        10468
4        10455
         ...  
66392    10301
66393    10302
66394    10302
66395    10310
66396    10306
Name: fips, Length: 66397, dtype: int64

In [ ]:
borough_to_county = {
    'BRONX': '005',
    'BROOKLYN': '047',
    'MANHATTAN': '061',
    'QUEENS': '081',
    'STATEN ISLAND': '085'
}

In [ ]:
merged_df['tract_str'] = (
    merged_df['census_tract']
    .astype(str)
    .str.replace('.0', '')
    .str.zfill(6)
)

In [ ]:
merged_df['tract_str'].head()

0    001152
1    000118
2    000122
3    000409
4    000035
Name: tract_str, dtype: object

In [ ]:
merged_df['rebuilt_fips'] = (
    '36' +
    merged_df['borough'].map(borough_to_county) +
    merged_df['tract_str']
)

In [ ]:
len(merged_df['rebuilt_fips'][0])

11

In [ ]:
svi_census_pop.fips

101     36005000100
102     36005000200
103     36005000400
104     36005001600
105     36005001901
           ...     
4282    36085030301
4283    36085030302
4284    36085031901
4285    36085031902
4286    36085032300
Name: fips, Length: 2324, dtype: int64

In [ ]:
# type(svi_census_pop['fips'])

In [ ]:
svi_census_pop['fips'].dtype

dtype('int64')

In [ ]:
type(svi_census_pop['fips'].iloc[0])

numpy.int64

In [ ]:
overlapping_tracts = set(merged_df['rebuilt_fips']).intersection(set(svi_census_pop['fips']))
num_overlapping = len(overlapping_tracts)
num_overlapping

0

In [ ]:
len(merged_df['rebuilt_fips'].unique())

2031

In [ ]:
len(svi_census_pop['fips'].unique())

2324

In [ ]:
# merged_df.rebuilt_fips[0].dtype()
type(merged_df['rebuilt_fips'].iloc[0])
merged_df['rebuilt_fips'] = merged_df['rebuilt_fips'].astype(int)

In [ ]:
overlapping_tracts = set(merged_df['rebuilt_fips']).intersection(set(svi_census_pop['fips']))
num_overlapping = len(overlapping_tracts)
num_overlapping

357

In [ ]:
len(set(merged_df['rebuilt_fips'])), num_overlapping/len(set(merged_df['rebuilt_fips']))
# only 17.58 overlapped fips

(2031, 0.1757754800590842)

## **conclusion: Rebuilding fips from eviction data is hard, because the reduilt_fips from borough, zipcodes, and census tract do not match well with all the census tracts and the real census tract from SVI**

# **Step 4: Normalize the eviction rates by nta population**

In [ ]:
merged_df.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

In [ ]:
merged_df.drop(columns = ['tract_str', 'rebuilt_fips'], inplace = True)

## **evictions_nta_per_1kpop**

In [ ]:
# nta_population = merged_df.groupby('nta')['e_totpop'].first().reset_index()
# nta_population

In [ ]:
# evictions_per_nta = merged_df.groupby('nta').size().reset_index(name='eviction_count')
# evictions_per_nta

In [ ]:
population_per_nta = merged_df.drop_duplicates('nta')[['nta', 'e_totpop']]
population_per_nta
# all rows here

nta  e_totpop
0                     East New York   96801.0
1                  Sunset Park East   93008.0
2                  Sunset Park West   93008.0
3             Van Cortlandt Village   81397.0
4            Mott Haven-Port Morris   44380.0
...                             ...       ...
8239              Rossville-Woodrow   34740.0
8344          Midtown-Midtown South   30930.0
9697                        Maspeth   39250.0
12075  Brooklyn Heights-Cobble Hill   69755.0
33471    park-cemetery-etc-Brooklyn   50319.0

[187 rows x 2 columns]

In [ ]:
merged_df['evictions_nta_per_1kpop'] = merged_df['average_year_eviction_nta_count'] / merged_df['e_totpop'] * 1000
merged_df[['bin', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False)

bin  evictions_nta_per_1kpop
29883  4032652                      inf
65395  4236378               181.927711
39043  1046407                36.375575
14692  1046407                36.375575
1756   1046407                36.375575
...        ...                      ...
30447  5079157                 0.053566
9663   5133499                 0.053566
66189  5105247                 0.053566
30665  5076332                 0.053566
33471  3054140                 0.003975

[66397 rows x 2 columns]

In [ ]:
merged_df.e_totpop.median()

71812.0

In [ ]:
bin_4032652	= merged_df[merged_df['bin'] == 4032652]
bin_4032652.nta
# pop: 69,529
# https://www.weichert.com/search/community/city.aspx?city=14635#:~:text=Total:%2069%2C529,Median%20Age:%200.00

29883    East Elmhurst
Name: nta, dtype: object

In [ ]:
merged_df.loc[merged_df['bin'] == 4032652, 'e_totpop'] = 69529

In [ ]:
merged_df['evictions_nta_per_1kpop'] = merged_df['average_year_eviction_nta_count'] / merged_df['e_totpop'] * 1000
merged_df[['bin','nta', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False).head(30)

bin                                nta  evictions_nta_per_1kpop
65395  4236378                         St. Albans               181.927711
14692  1046407        Lenox Hill-Roosevelt Island                36.375575
1756   1046407        Lenox Hill-Roosevelt Island                36.375575
39043  1046407        Lenox Hill-Roosevelt Island                36.375575
8027   2028351                  Crotona Park East                26.281454
23092  2002619   East Concourse-Concourse Village                25.496454
17201  2002619   East Concourse-Concourse Village                25.496454
63581  1083989  Central Harlem North-Polo Grounds                17.055074
63579  1053940  Central Harlem North-Polo Grounds                17.055074
63577  1053938  Central Harlem North-Polo Grounds                17.055074
63574  1083986  Central Harlem North-Polo Grounds                17.055074
9138   1083988  Central Harlem North-Polo Grounds                17.055074
9137   1083988  Central Harlem North-Polo Grounds                17.055074
9133   1054342  Central Harlem North-Polo Grounds                17.055074
9040   1054327  Central Harlem North-Polo Grounds                17.055074
9029   1054336  Central Harlem North-Polo Grounds                17.055074
63549  1053936  Central Harlem North-Polo Grounds                17.055074
7066   1053887  Central Harlem North-Polo Grounds                17.055074
7060   1053888  Central Harlem North-Polo Grounds                17.055074
7076   1081379  Central Harlem North-Polo Grounds                17.055074
63547  1053939  Central Harlem North-Polo Grounds                17.055074
63546  1053936  Central Harlem North-Polo Grounds                17.055074
63545  1053939  Central Harlem North-Polo Grounds                17.055074
63544  1083989  Central Harlem North-Polo Grounds                17.055074
9166   1060003  Central Harlem North-Polo Grounds                17.055074
9165   1053898  Central Harlem North-Polo Grounds                17.055074
63529  1053939  Central Harlem North-Polo Grounds                17.055074
63528  1083990  Central Harlem North-Polo Grounds                17.055074
63548  1053936  Central Harlem North-Polo Grounds                17.055074
6682   1053769  Central Harlem North-Polo Grounds                17.055074

In [ ]:
bin_4032652	= merged_df[merged_df['bin'] == 4032652]
bin_4032652.evictions_nta_per_1kpop
# good

29883    0.227243
Name: evictions_nta_per_1kpop, dtype: float64

In [ ]:
Albans = merged_df[merged_df['nta'] == 'St. Albans']
Albans.head(95)
# because this is done by zipcodes, so a mismatch of a very small zipcode and the very small population resulted in an outlier of
# normalized veictions per nta per 1k

primary_key         bbl court_index_number  docket_number  \
371    0304282/23_11772  4125290220         0304282/23          11772   
476    0308184/23_16419  4125290230         0308184/23          16419   
632    0315643/22_14225  4123900255         0315643/22          14225   
913      050639/20_2067  4110720034          050639/20           2067   
1069    054876/19_97546  4109870045          054876/19          97546   
...                 ...         ...                ...            ...   
28462  50138/18-2_25458  4109680136         50138/18-2          25458   
28751    50402/22_30618  4125380013           50402/22          30618   
28963     50639/20_2066  4110720034           50639/20           2066   
29045    50718/19_22722  4103530029           50718/19          22722   
29153    50836/20_29207  4103850037           50836/20          29207   

               eviction_address eviction_apartment_number executed_date  \
371    126-30 LOCUST MANOR LANE                        6C    2024-05-22   
476        169-30 BAISLEY BLVD.                        7A    2024-05-16   
632         172-25 BAISLEY BLVD                 1 FLR-RM1    2024-05-22   
913         200-16 116TH AVENUE                        #2    2023-03-15   
1069          191-19 113TH ROAD                         1    2019-10-16   
...                         ...                       ...           ...   
28462          190-62 112TH AVE                         2    2019-02-27   
28751         178-17 130TH AVE.                   unknown    2023-09-05   
28963       200-16 116TH AVENUE                        #2    2023-03-15   
29045   183-03 HENDERSON AVENUE                   unknown    2019-05-14   
29153      116-19 MEXICO STREET                (BASEMENT)    2023-09-25   

      borough  zipcode         ejectment eviction/legal_possession   latitude  \
371    QUEENS    11434  Not an Ejectment                Possession  40.680366   
476    QUEENS    11434  Not an Ejectment                Possession  40.680784   
632    QUEENS    11434  Not an Ejectment                Possession  40.682099   
913    QUEENS    11412  Not an Ejectment                Possession  40.696003   
1069   QUEENS    11412  Not an Ejectment                Possession  40.698780   
...       ...      ...               ...                       ...        ...   
28462  QUEENS    11412  Not an Ejectment                Possession  40.700663   
28751  QUEENS    11434  Not an Ejectment                Possession  40.680077   
28963  QUEENS    11412  Not an Ejectment                Possession  40.696003   
29045  QUEENS    11423  Not an Ejectment                Possession  40.706804   
29153  QUEENS    11412  Not an Ejectment                Possession  40.693460   

       longitude  community_board  council_district  census_tract      bin  \
371   -73.772190             12.0              27.0       33401.0  4596156   
476   -73.773998             12.0              27.0       33401.0  4542666   
632   -73.771549             12.0              27.0         282.0  4268501   
913   -73.752630             12.0              27.0         530.0  4238511   
1069  -73.762335             12.0              27.0         524.0  4235647   
...          ...              ...               ...           ...      ...   
28462 -73.763703             12.0              27.0         524.0  4235067   
28751 -73.763725             12.0              27.0       33401.0  4270385   
28963 -73.752630             12.0              27.0         530.0  4238511   
29045 -73.774151             12.0              27.0         404.0  4220575   
29153 -73.763714             12.0              27.0         398.0  4221093   

              nta  year month_year                      geometry  yearbuilt  \
371    St. Albans  2024    2024-05   POINT (-73.77219 40.680366)     2018.0   
476    St. Albans  2024    2024-05  POINT (-73.773998 40.680784)     2013.0   
632    St. Albans  2024    2024-05  POINT (-73.771549 40.682099)     1910.0   
913    S

In [ ]:
median_of_Albans = Albans['evictions_nta_per_1kpop'].median()
median_of_Albans

1.4879412706624295

In [ ]:
merged_df.loc[merged_df['bin'] == 4236378, 'evictions_nta_per_1kpop'] = median_of_Albans

In [ ]:
merged_df[['bin','nta', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False).head(30)

bin                                nta  evictions_nta_per_1kpop
14692  1046407        Lenox Hill-Roosevelt Island                36.375575
1756   1046407        Lenox Hill-Roosevelt Island                36.375575
39043  1046407        Lenox Hill-Roosevelt Island                36.375575
8027   2028351                  Crotona Park East                26.281454
17201  2002619   East Concourse-Concourse Village                25.496454
23092  2002619   East Concourse-Concourse Village                25.496454
8317   1053939  Central Harlem North-Polo Grounds                17.055074
8449   1083988  Central Harlem North-Polo Grounds                17.055074
6753   1083988  Central Harlem North-Polo Grounds                17.055074
6722   1083990  Central Harlem North-Polo Grounds                17.055074
9769   1053940  Central Harlem North-Polo Grounds                17.055074
9765   1083990  Central Harlem North-Polo Grounds                17.055074
9764   1053936  Central Harlem North-Polo Grounds                17.055074
9229   1054340  Central Harlem North-Polo Grounds                17.055074
6738   1053895  Central Harlem North-Polo Grounds                17.055074
6940   1083990  Central Harlem North-Polo Grounds                17.055074
6939   1053890  Central Harlem North-Polo Grounds                17.055074
6937   1053939  Central Harlem North-Polo Grounds                17.055074
6936   1053887  Central Harlem North-Polo Grounds                17.055074
6935   1053888  Central Harlem North-Polo Grounds                17.055074
9590   1053892  Central Harlem North-Polo Grounds                17.055074
6703   1053898  Central Harlem North-Polo Grounds                17.055074
6698   1053892  Central Harlem North-Polo Grounds                17.055074
6697   1053898  Central Harlem North-Polo Grounds                17.055074
6696   1053909  Central Harlem North-Polo Grounds                17.055074
6695   1053939  Central Harlem North-Polo Grounds                17.055074
8824   1053943  Central Harlem North-Polo Grounds                17.055074
9752   1053938  Central Harlem North-Polo Grounds                17.055074
6723   1053890  Central Harlem North-Polo Grounds                17.055074
6737   1053890  Central Harlem North-Polo Grounds                17.055074

In [ ]:
merged_df.evictions_nta_per_1kpop.describe(), merged_df.evictions_nta_per_1kpop.median()

(count    66397.000000
 mean         2.299831
 std          2.076064
 min          0.003975
 25%          1.150948
 50%          1.898226
 75%          2.960793
 max         36.375575
 Name: evictions_nta_per_1kpop, dtype: float64,
 1.8982260779489835)

$\text{evictions_nta_per_1kpop} = \left( \frac{\text{average_year_eviction_nta_count}}{\text{e_totpop}} \right) \times 1000$

## **Interpretations**

- represents the average annual number of evictions in an nta normalized by the nta's population.
- for example: if evicions_nta_per_1kpop = 120, the NTA had 120 evictions per year on average per 1000 people.

# **evictions_nta_per_1kunit_per_1kpop**

## **Add buildings_affected column**

In [ ]:
evictions_per_nta = merged_df.groupby('nta')['bin'].nunique().reset_index(name='buildings_affected_per_nta')
evictions_per_nta.head()

nta  buildings_affected_per_nta
0                     Allerton-Pelham Gardens                          97
1  Annadale-Huguenot-Prince's Bay-Eltingville                          15
2                               Arden Heights                          16
3                                     Astoria                         267
4                                  Auburndale                          24

In [ ]:
# unique_buildings = normal.drop_duplicates(subset=['nta', 'bin'])
# unique_buildings.head()

In [ ]:
# total_units_per_nta = unique_buildings.groupby('nta')['unitsres'].sum().reset_index(name='total_units_per_nta')
# total_units_per_nta.head()

In [ ]:
# building_units_per_nta = pd.merge(evictions_per_nta, total_units_per_nta, on='nta', how='left')
# building_units_per_nta.head()

In [ ]:
merged_df = merged_df.merge(evictions_per_nta, on='nta', how='left')
merged_df.shape
# correct, now, buildings_affected column has been added.

(66397, 96)

In [ ]:
merged_df.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

In [ ]:
# merged_df.drop(columns = ['buildings_affected_per_nta_y'], inplace = True)
# merged_df.rename(columns = {'buildings_affected_per_nta_x': 'buildings_affected_per_nta'}, inplace = True)

# **Add units_per_nta**

In [ ]:
merged_df['unit_per_nta'] = merged_df['unitsres'] * merged_df['buildings_affected_per_nta']
merged_df[['bin', 'unit_per_nta']].head()

bin  unit_per_nta
0  3083989        2175.0
1  3143881         272.0
2  3143435         588.0
3  2015444       85888.0
4  2003900       20995.0

In [ ]:
merged_df[['bin', 'nta','average_year_eviction_count', 'average_year_eviction_unit_count']].head(10)
# for the same nta, they likely have different values for these two columns, because they were groupedby bin
# but for the same bin, they have the same values for these two columns, because they were groupedby bin

bin                         nta  average_year_eviction_count  \
0  3083989               East New York                          0.8   
1  3143881            Sunset Park East                          0.6   
2  3143435            Sunset Park West                          0.6   
3  2015444       Van Cortlandt Village                          0.8   
4  2003900      Mott Haven-Port Morris                          1.6   
5  2013945  Bedford Park-Fordham North                          1.6   
6  2017108  Bedford Park-Fordham North                          1.0   
7  2011545          Claremont-Bathgate                          1.6   
8  3089943               East New York                          0.4   
9  3245453              Brighton Beach                          1.2   

   average_year_eviction_unit_count  
0                          0.266667  
1                          0.300000  
2                          0.150000  
3                          0.002273  
4                          0.024615  
5                          0.038095  
6                          0.500000  
7                          0.048485  
8                          0.080000  
9                          0.007692

sum all evictions per NTA -> divide by years -> average_year_eviction_nta_count. \
sum all units per NTA (deduplicate buildings to avoid overcounting). \
divide average_year_eviction_nta_count by total units -> evictions per unit per year. \
divide by NTA population and multiply by 1,000 -> evictions per unit per 1k residents. \

In [ ]:
# merged_df['e_totpop'].head()

In [ ]:
merged_df.columns, merged_df.shape

(Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
        'latitude', 'longitude', 'community_board', 'council_district',
        'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
        'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
        'bldgarea', 'building_type', 'building_category', 'is_condo',
        'floor_category', 'rent_era', 'architectural_style', 'economic_period',
        'residential_units_category', 'is_llc', 'building_size_category',
        'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
        'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
        'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
        'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
        'ep_hisp', 'ep_asian', 'ep_aian', 'ep_n

In [ ]:
merged_df['evictions_nta_per_1kunit_per_1kpop'] = \
      merged_df['average_year_eviction_nta_count'] / (merged_df['unit_per_nta']*merged_df['e_totpop']) * 1000 * 1000

In [ ]:
merged_df[['bin', 'nta', 'evictions_nta_per_1kunit_per_1kpop']].sort_values('evictions_nta_per_1kunit_per_1kpop', ascending = False)

bin                                  nta  \
65395  4236378                           St. Albans   
54507  2082091  Pelham Bay-Country Club-City Island   
25274  2090621  Pelham Bay-Country Club-City Island   
23286  2006658                          Hunts Point   
12979  2082172  Pelham Bay-Country Club-City Island   
...        ...                                  ...   
9272   1082726       Stuyvesant Town-Cooper Village   
41666  1082885       Stuyvesant Town-Cooper Village   
47187  1082883       Stuyvesant Town-Cooper Village   
45357  1082869       Stuyvesant Town-Cooper Village   
42860  1082880       Stuyvesant Town-Cooper Village   

       evictions_nta_per_1kunit_per_1kpop  
65395                          427.060354  
54507                           83.731361  
25274                           83.731361  
23286                           44.748396  
12979                           41.865680  
...                                   ...  
9272                             0.000665  
41666                            0.000665  
47187                            0.000665  
45357                            0.000665  
42860                            0.000665  

[66397 rows x 3 columns]

## **Interpretations:**
- (evictions per nta/ population per nta) * 1,000 * 1,000 = Evictions per 1k units per 1k people. [note, intentially standardized twice over population and units]
- this formula ensures eviction rates account for housing stock (units) and population.
- for example: 427.060354 means 427.060354 evictions per 1,000 units per 1,000 people in St.Albans.
- Advantage: compare neighborhoods fairly (e.g., St.Albans has higher pressure, 7.39, than Stuyvesant Town-Cooper Village).
- The formula:

$\text{Rate} = \left( \frac{\text{Total Evictions}(\text{neighborhood})}{\text{Total Units}(\text{neighborhood}) \times \text{Total Population}(\text{neighborhood})} \right) \times 1,\!000 \times 1,\!000$

In [ ]:
merged_df[['nta', 'average_year_eviction_count', 'average_year_eviction_unit_count',
           'average_year_eviction_nta_count','evictions_nta_per_1kpop',
           'evictions_nta_per_1kunit_per_1kpop']].sort_values(by = 'evictions_nta_per_1kunit_per_1kpop', ascending = False)

nta  average_year_eviction_count  \
65395                           St. Albans                          0.4   
54507  Pelham Bay-Country Club-City Island                          0.2   
25274  Pelham Bay-Country Club-City Island                          0.2   
23286                          Hunts Point                          0.2   
12979  Pelham Bay-Country Club-City Island                          0.2   
...                                    ...                          ...   
9272        Stuyvesant Town-Cooper Village                          0.2   
41666       Stuyvesant Town-Cooper Village                          0.4   
47187       Stuyvesant Town-Cooper Village                          0.2   
45357       Stuyvesant Town-Cooper Village                          0.4   
42860       Stuyvesant Town-Cooper Village                          0.2   

       average_year_eviction_unit_count  average_year_eviction_nta_count  \
65395                          0.200000                             60.4   
54507                          0.200000                             46.0   
25274                          0.200000                             46.0   
23286                          0.200000                            106.0   
12979                          0.100000                             46.0   
...                                 ...                              ...   
9272                           0.000023                             12.6   
41666                          0.000046                             12.6   
47187                          0.000023                             12.6   
45357                          0.000046                             12.6   
42860                          0.000023                             12.6   

       evictions_nta_per_1kpop  evictions_nta_per_1kunit_per_1kpop  
65395                 1.487941                          427.060354  
54507                10.717614                           83.731361  
25274                10.717614                           83.731361  
23286                 9.397163                           44.748396  
12979                10.717614                           41.865680  
...                        ...                                 ...  
9272                  0.215687                            0.000665  
41666                 0.215687                            0.000665  
47187                 0.215687                            0.000665  
45357                 0.215687                            0.000665  
42860                 0.215687                            0.000665  

[66397 rows x 6 columns]

In [ ]:
merged_df.shape
# 98 is good, 96 + uniter per nta, building per nta

(66397, 98)

In [ ]:
# merged_df.drop(columns = ['evictions_nta_per_unit_per_1k'], inplace = True)

In [ ]:
merged_df.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

In [ ]:
merged_df.isna().sum().sum()
# good

np.int64(0)

In [ ]:
# merged_df.drop(columns = ['evictions_nta_per_1k_y'], inplace = True)
# merged_df.rename(columns = {'evictions_nta_per_1k_x': 'evictions_nta_per_1k'}, inplace = True)

# **Step 5 Normalize eviction rates per nta population for Covid**

In [ ]:
link4 = '/content/drive/My Drive/X999/bbl_evictions_311_svi_covid.csv'
merged_df_covid = pd.read_csv(link4)
merged_df_covid.shape

(5386, 93)

In [ ]:
merged_df_covid.isna().sum().sum()

np.int64(0)

In [ ]:
merged_df_covid.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

## **evictions_nta_per_1kpop**

In [ ]:
population_per_nta = merged_df_covid.drop_duplicates('nta')[['nta', 'e_totpop']]
population_per_nta
# all rows here

nta  e_totpop
0                            Kingsbridge Heights   81397.0
1                                   Forest Hills   75212.0
2                                        Erasmus  101053.0
3                            Crown Heights South   58476.0
4                                    Kew Gardens   20315.0
...                                          ...       ...
3141                  Jamaica Estates-Holliswood   64268.0
3370                                  Auburndale   37647.0
3430              Stuyvesant Town-Cooper Village   32410.0
4746                               Starrett City   16808.0
5312  Annadale-Huguenot-Prince's Bay-Eltingville   34740.0

[185 rows x 2 columns]

In [ ]:
merged_df_covid['evictions_nta_per_1kpop'] = merged_df_covid['average_year_eviction_nta_count'] / merged_df_covid['e_totpop'] * 1000
merged_df_covid[['bin', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False)

bin  evictions_nta_per_1kpop
3110  1046407                 7.441453
915   1013698                 2.386727
982   1013698                 2.386727
1549  1013633                 2.386727
3760  1000793                 1.962884
...       ...                      ...
3074  4296569                 0.010974
5312  5082694                 0.009595
3370  4124595                 0.008854
2516  2121023                 0.006754
2515  2121023                 0.006754

[5386 rows x 2 columns]

In [ ]:
merged_df_covid['evictions_nta_per_1kpop'] = merged_df_covid['average_year_eviction_nta_count'] / merged_df_covid['e_totpop'] * 1000
merged_df_covid[['bin','nta', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False).head(30)

bin                                         nta  \
3110  1046407                 Lenox Hill-Roosevelt Island   
915   1013698                                     Clinton   
982   1013698                                     Clinton   
1549  1013633                                     Clinton   
3760  1000793           Battery Park City-Lower Manhattan   
803   1000793           Battery Park City-Lower Manhattan   
763   1000810           Battery Park City-Lower Manhattan   
1067  4541713        Hunters Point-Sunnyside-West Maspeth   
3407  4541713        Hunters Point-Sunnyside-West Maspeth   
3522  4000019        Hunters Point-Sunnyside-West Maspeth   
1097  4541713        Hunters Point-Sunnyside-West Maspeth   
973   4541715        Hunters Point-Sunnyside-West Maspeth   
1291  1012983  Hudson Yards-Chelsea-Flatiron-Union Square   
5146  1013587  Hudson Yards-Chelsea-Flatiron-Union Square   
3850  1087624  Hudson Yards-Chelsea-Flatiron-Union Square   
3701  1083613  Hudson Yards-Chelsea-Flatiron-Union Square   
3786  1088724  Hudson Yards-Chelsea-Flatiron-Union Square   
1436  1013602  Hudson Yards-Chelsea-Flatiron-Union Square   
2854  5165244  Hudson Yards-Chelsea-Flatiron-Union Square   
4725  1053936           Central Harlem North-Polo Grounds   
429   1087984           Central Harlem North-Polo Grounds   
425   1083988           Central Harlem North-Polo Grounds   
424   1083988           Central Harlem North-Polo Grounds   
420   1053890           Central Harlem North-Polo Grounds   
418   1053940           Central Harlem North-Polo Grounds   
419   1053890           Central Harlem North-Polo Grounds   
422   1053943           Central Harlem North-Polo Grounds   
5120  1053940           Central Harlem North-Polo Grounds   
5123  1053914           Central Harlem North-Polo Grounds   
4723  1053837           Central Harlem North-Polo Grounds   

      evictions_nta_per_1kpop  
3110                 7.441453  
915                  2.386727  
982                  2.386727  
1549                 2.386727  
3760                 1.962884  
803                  1.962884  
763                  1.962884  
1067                 1.923077  
3407                 1.923077  
3522                 1.923077  
1097                 1.923077  
973                  1.923077  
1291                 1.501328  
5146                 1.501328  
3850                 1.501328  
3701                 1.501328  
3786                 1.501328  
1436                 1.501328  
2854                 1.501328  
4725                 1.410806  
429                  1.410806  
425                  1.410806  
424                  1.410806  
420                  1.410806  
418                  1.410806  
419                  1.410806  
422                  1.410806  
5120                 1.410806  
5123                 1.410806  
4723                 1.410806

In [ ]:
merged_df_covid[['bin','nta', 'evictions_nta_per_1kpop']].sort_values(by = 'evictions_nta_per_1kpop', ascending = False).head(30)

bin                                         nta  \
3110  1046407                 Lenox Hill-Roosevelt Island   
915   1013698                                     Clinton   
982   1013698                                     Clinton   
1549  1013633                                     Clinton   
3760  1000793           Battery Park City-Lower Manhattan   
803   1000793           Battery Park City-Lower Manhattan   
763   1000810           Battery Park City-Lower Manhattan   
1067  4541713        Hunters Point-Sunnyside-West Maspeth   
3407  4541713        Hunters Point-Sunnyside-West Maspeth   
3522  4000019        Hunters Point-Sunnyside-West Maspeth   
1097  4541713        Hunters Point-Sunnyside-West Maspeth   
973   4541715        Hunters Point-Sunnyside-West Maspeth   
1291  1012983  Hudson Yards-Chelsea-Flatiron-Union Square   
5146  1013587  Hudson Yards-Chelsea-Flatiron-Union Square   
3850  1087624  Hudson Yards-Chelsea-Flatiron-Union Square   
3701  1083613  Hudson Yards-Chelsea-Flatiron-Union Square   
3786  1088724  Hudson Yards-Chelsea-Flatiron-Union Square   
1436  1013602  Hudson Yards-Chelsea-Flatiron-Union Square   
2854  5165244  Hudson Yards-Chelsea-Flatiron-Union Square   
4725  1053936           Central Harlem North-Polo Grounds   
429   1087984           Central Harlem North-Polo Grounds   
425   1083988           Central Harlem North-Polo Grounds   
424   1083988           Central Harlem North-Polo Grounds   
420   1053890           Central Harlem North-Polo Grounds   
418   1053940           Central Harlem North-Polo Grounds   
419   1053890           Central Harlem North-Polo Grounds   
422   1053943           Central Harlem North-Polo Grounds   
5120  1053940           Central Harlem North-Polo Grounds   
5123  1053914           Central Harlem North-Polo Grounds   
4723  1053837           Central Harlem North-Polo Grounds   

      evictions_nta_per_1kpop  
3110                 7.441453  
915                  2.386727  
982                  2.386727  
1549                 2.386727  
3760                 1.962884  
803                  1.962884  
763                  1.962884  
1067                 1.923077  
3407                 1.923077  
3522                 1.923077  
1097                 1.923077  
973                  1.923077  
1291                 1.501328  
5146                 1.501328  
3850                 1.501328  
3701                 1.501328  
3786                 1.501328  
1436                 1.501328  
2854                 1.501328  
4725                 1.410806  
429                  1.410806  
425                  1.410806  
424                  1.410806  
420                  1.410806  
418                  1.410806  
419                  1.410806  
422                  1.410806  
5120                 1.410806  
5123                 1.410806  
4723                 1.410806

In [ ]:
merged_df_covid.evictions_nta_per_1kpop.describe(), merged_df_covid.evictions_nta_per_1kpop.median()

(count    5386.000000
 mean        0.285140
 std         0.226549
 min         0.006754
 25%         0.148356
 50%         0.240340
 75%         0.376040
 max         7.441453
 Name: evictions_nta_per_1kpop, dtype: float64,
 0.24034017378443334)

In [ ]:
merged_df_covid.isna().sum().sum()

np.int64(0)

$\text{evictions_nta_per_1kpop} = \left( \frac{\text{average_year_eviction_nta_count}}{\text{e_totpop}} \right) \times 1000$

## **Interpretations**

- represents the average annual number of evictions in an nta normalized by the nta's population.
- for example: if evicions_nta_per_1k = 120, the NTA had 120 evictions per year on average per 1000 people.

# **evictions_nta_per_1kunit_per_1kpop**

## **Add buildings_affected column**

In [ ]:
evictions_per_nta = merged_df_covid.groupby('nta')['bin'].nunique().reset_index(name='buildings_affected_per_nta')
evictions_per_nta.head()

nta  buildings_affected_per_nta
0                     Allerton-Pelham Gardens                          12
1  Annadale-Huguenot-Prince's Bay-Eltingville                           1
2                               Arden Heights                           5
3                                     Astoria                          18
4                                  Auburndale                           1

In [ ]:
merged_df_covid = merged_df_covid.merge(evictions_per_nta, on='nta', how='left')
merged_df_covid.shape
# correct, now, buildings_affected column has been added.

(5386, 95)

In [ ]:
merged_df_covid.columns

Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
       'eviction_address', 'eviction_apartment_number', 'executed_date',
       'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
       'latitude', 'longitude', 'community_board', 'council_district',
       'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
       'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
       'bldgarea', 'building_type', 'building_category', 'is_condo',
       'floor_category', 'rent_era', 'architectural_style', 'economic_period',
       'residential_units_category', 'is_llc', 'building_size_category',
       'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
       'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
       'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
       'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
       'ep_hisp', 'ep_asian', 'ep_aian', 'ep_nhpi', 'ep_twom

# **Add units_per_nta**

In [ ]:
merged_df_covid['unit_per_nta'] = merged_df_covid['unitsres'] * merged_df_covid['buildings_affected_per_nta']
merged_df_covid[['bin', 'unit_per_nta']].head()

bin  unit_per_nta
0  2113173         114.0
1  4074666        3077.0
2  3117969          92.0
3  3029673        2064.0
4  4079390         948.0

In [ ]:
merged_df_covid[['bin', 'nta','average_year_eviction_count', 'average_year_eviction_unit_count']].head(10)
# for the same nta, they likely have different values for these two columns, because they were groupedby bin
# but for the same bin, they have the same values for these two columns, because they were groupedby bin

bin                               nta  average_year_eviction_count  \
0  2113173               Kingsbridge Heights                          0.2   
1  4074666                      Forest Hills                          0.2   
2  3117969                           Erasmus                          0.6   
3  3029673               Crown Heights South                          0.8   
4  4079390                       Kew Gardens                          0.4   
5  1063219          Washington Heights South                          0.4   
6  2003313                        Highbridge                          0.8   
7  3337046                     East New York                          0.6   
8  2126620  East Concourse-Concourse Village                          0.6   
9  3324603               Crown Heights North                          0.2   

   average_year_eviction_unit_count  
0                          0.066667  
1                          0.001105  
2                          0.300000  
3                          0.016667  
4                          0.005063  
5                          0.013333  
6                          0.002827  
7                          0.000416  
8                          0.002765  
9                          0.000171

sum all evictions per NTA -> divide by years -> average_year_eviction_nta_count. \
sum all units per NTA (deduplicate buildings to avoid overcounting). \
divide average_year_eviction_nta_count by total units -> evictions per unit per year. \
divide by NTA population and multiply by 1,000 -> evictions per unit per 1k residents. \

In [ ]:
# merged_df['e_totpop'].head()

In [ ]:
merged_df_covid.columns, merged_df_covid.shape

(Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
        'latitude', 'longitude', 'community_board', 'council_district',
        'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
        'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
        'bldgarea', 'building_type', 'building_category', 'is_condo',
        'floor_category', 'rent_era', 'architectural_style', 'economic_period',
        'residential_units_category', 'is_llc', 'building_size_category',
        'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
        'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
        'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
        'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
        'ep_hisp', 'ep_asian', 'ep_aian', 'ep_n

In [ ]:
merged_df_covid['evictions_nta_per_1kunit_per_1kpop'] = \
      merged_df_covid['average_year_eviction_nta_count'] / (merged_df_covid['unit_per_nta']*merged_df_covid['e_totpop']) * 1000 * 1000

In [ ]:
merged_df_covid[['bin', 'nta', 'evictions_nta_per_1kunit_per_1kpop']].sort_values('evictions_nta_per_1kunit_per_1kpop', ascending = False)

bin                                     nta  \
2446  2082724     Pelham Bay-Country Club-City Island   
2444  2082048     Pelham Bay-Country Club-City Island   
2633  5116366  Charleston-Richmond Valley-Tottenville   
807   5088263  Charleston-Richmond Valley-Tottenville   
2586  5023744                           Port Richmond   
...       ...                                     ...   
504   2128845                              Co-op City   
2279  2124552                              Co-op City   
2278  2095393                              Co-op City   
2277  2095397                              Co-op City   
2276  2128848                              Co-op City   

      evictions_nta_per_1kunit_per_1kpop  
2446                           88.758709  
2444                           44.379355  
2633                           22.591212  
807                            22.591212  
2586                           20.945166  
...                                  ...  
504                             0.000786  
2279                            0.000786  
2278                            0.000786  
2277                            0.000786  
2276                            0.000786  

[5386 rows x 3 columns]

## **Interpretations:**
- (evictions per nta/ population per nta) * 1,000 * 1,000 = Evictions per 1k units per 1k people. [note, intentially standardized twice over population and units]
- this formula ensures eviction rates account for housing stock (units) and population.
- for example: 427.060354 means 427.060354 evictions per 1,000 units per 1,000 people in St.Albans.
- Advantage: compare neighborhoods fairly (e.g., St.Albans has higher pressure, 7.39, than Stuyvesant Town-Cooper Village).
- The formula:

$\text{Rate} = \left( \frac{\text{Total Evictions}(\text{neighborhood})}{\text{Total Units}(\text{neighborhood}) \times \text{Total Population}(\text{neighborhood})} \right) \times 1,\!000 \times 1,\!000$

In [ ]:
merged_df_covid[['nta', 'average_year_eviction_count', 'average_year_eviction_unit_count',
           'average_year_eviction_nta_count','evictions_nta_per_1kpop',
           'evictions_nta_per_1kunit_per_1kpop']].sort_values(by = 'evictions_nta_per_1kunit_per_1kpop', ascending = False)

nta  average_year_eviction_count  \
2446     Pelham Bay-Country Club-City Island                          0.2   
2444     Pelham Bay-Country Club-City Island                          0.2   
2633  Charleston-Richmond Valley-Tottenville                          0.2   
807   Charleston-Richmond Valley-Tottenville                          0.2   
2586                           Port Richmond                          0.2   
...                                      ...                          ...   
504                               Co-op City                          0.2   
2279                              Co-op City                          0.2   
2278                              Co-op City                          0.4   
2277                              Co-op City                          0.2   
2276                              Co-op City                          0.2   

      average_year_eviction_unit_count  average_year_eviction_nta_count  \
2446                          0.200000                         5.333333   
2444                          0.100000                         5.333333   
2633                          0.200000                         1.666667   
807                           0.200000                         1.666667   
2586                          0.200000                         4.666667   
...                                ...                              ...   
504                           0.000018                         9.333333   
2279                          0.000018                         9.333333   
2278                          0.000037                         9.333333   
2277                          0.000018                         9.333333   
2276                          0.000018                         9.333333   

      evictions_nta_per_1kpop  evictions_nta_per_1kunit_per_1kpop  
2446                 1.242622                           88.758709  
2444                 1.242622                           44.379355  
2633                 0.112956                           22.591212  
807                  0.112956                           22.591212  
2586                 0.251342                           20.945166  
...                       ...                                 ...  
504                  0.214476                            0.000786  
2279                 0.214476                            0.000786  
2278                 0.214476                            0.000786  
2277                 0.214476                            0.000786  
2276                 0.214476                            0.000786  

[5386 rows x 6 columns]

In [ ]:
merged_df_covid.shape, merged_df.shape
# 97 and 98 is good, 95/96 + uniter per nta, building per nta

((5386, 97), (66397, 98))

In [ ]:
# merged_df.drop(columns = ['evictions_nta_per_1k'], inplace = True)

In [ ]:
merged_df_covid.isna().sum().sum(), merged_df.columns

(np.int64(0),
 Index(['primary_key', 'bbl', 'court_index_number', 'docket_number',
        'eviction_address', 'eviction_apartment_number', 'executed_date',
        'borough', 'zipcode', 'ejectment', 'eviction/legal_possession',
        'latitude', 'longitude', 'community_board', 'council_district',
        'census_tract', 'bin', 'nta', 'year', 'month_year', 'geometry',
        'yearbuilt', 'bldgclass', 'numfloors', 'unitsres', 'ownername',
        'bldgarea', 'building_type', 'building_category', 'is_condo',
        'floor_category', 'rent_era', 'architectural_style', 'economic_period',
        'residential_units_category', 'is_llc', 'building_size_category',
        'size_quartile', 'decade', 'fips', 'e_totpop', 'rpl_theme1',
        'rpl_theme2', 'rpl_theme3', 'rpl_theme4', 'rpl_themes', 'ep_pov150',
        'ep_unemp', 'ep_nohsdp', 'ep_uninsur', 'ep_age65', 'ep_age17',
        'ep_disabl', 'ep_limeng', 'ep_noveh', 'ep_crowd', 'ep_hburd', 'ep_afam',
        'ep_hisp', 'ep_asian', 'e

In [ ]:
# save the correct coordinates' df to the drive:
merged_df.to_csv('/content/drive/My Drive/X999/bbl_evictions_311_svi_normal_times_2.7.csv', index=False)
merged_df_covid.to_csv('/content/drive/My Drive/X999/bbl_evictions_311_svi_covid_2.7.csv', index=False)